R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -332,7 +332,24 @@ test_that("createDataFrame Arrow optimization", {
   })
 })
 
-test_that("createDataFrame Arrow optimization - type specification", {
+test_that("createDataFrame/collect Arrow optimization - many partitions (partition order test)", {
+  skip_if_not_installed("arrow")
+
+  conf <- callJMethod(sparkSession, "conf")
+  arrowEnabled <- sparkR.conf("spark.sql.execution.arrow.enabled")[[1]]
+
+  callJMethod(conf, "set", "spark.sql.execution.arrow.enabled", "true")
+  tryCatch({
+    expect_equal(collect(createDataFrame(mtcars, numPartitions = 32)),
```

Comment:
I don't know if I'd say 32 is a large number of partitions, but should be good enough to verify the collect is working so I think it will be fine.

Reply:
They are evenly same 1 row per partitions so it wouldn't be likely in an order anyway .. I also manually verifies that it comes out of order pretty consistently.